### ISSUE TITLE
(0.18.0) Listing type on front page does not honor setting for non-logged in users

### ISSUE THREAD

**Issue by chenew** [LemmyNet/lemmy-ui#1612]
### Requirements

- [X] This is a bug report, and if not, please post to https://lemmy.ml/c/lemmy_support instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this issue already exists.
- [X] It's a single bug. Do not report multiple bugs in one issue.
- [X] It's a frontend issue, not a backend issue; Otherwise please create an issue on the [backend repo](https://github.com/LemmyNet/lemmy) instead.

### Summary

Since upgrading to 0.18.0, the listing type on the front page is defaulting to Local, even though the site setting is set to display All.

Set listing type to All in site setting:
<img width="139" alt="image" src="https://github.com/LemmyNet/lemmy-ui/assets/5931313/bae4a4e7-ae82-4226-be97-7c59d98f3b9a">

What a non-logged in user sees:
<img width="201" alt="image" src="https://github.com/LemmyNet/lemmy-ui/assets/5931313/b2cc84a8-76b8-41f6-abcd-b552e42b291e">

Setting the listing type to 'All' helps smaller instances that may not have a lot of posts.

### Steps to Reproduce

1. In site setting, set the Listing Type setting to 'All'
2. Visit the site as a non-logged in user
3. The listing type will default to 'Local'
4. Press F12 to bring up browser developer tools and hold the reload button to empty cache and hard reload

### Technical Details

OS: Ubuntu 22.04
Browser: Tried on latest versions of Brave, Edge, Chrome

### Lemmy Instance Version

0.18.0

**Comment by randommouse** [comment]
Can confirm the same results with Firefox.

**Comment by delendum** [comment]
Confirming this is still an issue in UI: 0.18.1-rc.7

**Comment by bdonvr** [comment]
Yes, as a small instance admin this is quite frustrating. I imagine I (and possibly Lemmy as a whole) have lost a non-trivial amount of potential users because they thought the local feed was all that was going on Lemmy.

It seems @jcgurango solved this in an earlier version of Lemmy-UI, on a fork they made as a "hotfix". However the file he changed has been significantly updated and I don't have the skills to integrate his fix into latest. https://github.com/jcgurango/lemmy-ui/commit/a360c9a3d83e91a0a47d5633b1d4e5ecf5a964df

**Comment by jcgurango** [comment]
@bdonvr just got around to updating my own small instance today (halubilo.social). I've made a docker image people can try out based on 0.18.1-rc.7 if they want: `jcgurango/lemmy-ui:0.18.1-rc.7-hotfix` and the code is up on the PR branch for those that wanna build it themselves.

**Comment by chenew** [comment]
This is now fixed with 0.18.2 release. Thanks @jcgurango for the fix!
